crates/biome_formatter/src/printer/mod.rs

```diff
@@ -105,29 +105,38 @@ impl<'a> Printer<'a> {
             } => self.print_text(slice, Some(*source_position)),
 
             FormatElement::Line(line_mode) => {
-                if args.mode().is_flat()
-                    && matches!(line_mode, LineMode::Soft | LineMode::SoftOrSpace)
-                {
-                    if line_mode == &LineMode::SoftOrSpace && self.state.line_width > 0 {
-                        self.state.pending_space = true;
+                if args.mode().is_flat() {
+                    match line_mode {
+                        LineMode::Soft | LineMode::SoftOrSpace => {
+                            if line_mode == &LineMode::SoftOrSpace && self.state.line_width > 0 {
+                                self.state.pending_space = true;
+                            }
+                            return Ok(());
+                        }
+                        LineMode::Hard | LineMode::Empty => {
+                            self.state.measured_group_fits = false;
+                        }
```

Comment:
This is the only actual change in the whole `FormatElement::Line` match arm. The rest is just refactoring to make sure this case gets hit at the right time, and the diff is just whitespace